ffi/src/handle.rs

```diff
@@ -172,3 +173,74 @@ macro_rules! gen_ffi_impl {
         }
     };
 }
+
+/// Generate FFI equivalent implementation of the requested trait method (e.g. Clone, Eq, Ord)
+#[macro_export]
+#[cfg(feature = "client")]
+macro_rules! gen_ffi_impl {
+    ( $vis:vis Clone: $( $other:ty ),+ $(,)? ) => {
+        extern {
+            /// FFI function equivalent of [`Clone::clone`]
+            ///
+            /// # Safety
+            ///
+            /// All of the given pointers must be valid and the given handle id must match the expected
+            /// pointer type
+            #[no_mangle]
+            $vis fn __clone(
+                handle_id: $crate::handle::Id,
+                handle_ptr: *const core::ffi::c_void,
+                output_ptr: *mut *mut core::ffi::c_void
+            ) -> $crate::FfiResult;
```

Comment:
So this is just a function declaration, but there is the actual definition/implementation? Macro name and doc says that there should be an implementation

Reply:
fixed the docs